llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -824,6 +813,55 @@ void InterBlockScheduling::emitInterBlockTop(const BlockState &BS) const {
   }
 }
 
+void InterBlockScheduling::emitInterBlockTop(BlockState &BS) {
+  if (BS.Kind != BlockType::Epilogue) {
+    return;
+  }
+
+  MachineBasicBlock *BB = BS.TheBlock;
+  MachineBasicBlock *Loop = AIELoopUtils::getLoopPredecessor(*BB);
+  assert(Loop);
+  const BlockState &LBS = getBlockState(Loop);
+
+  // Emit the bundles of the swp epilog in a dedicated exit.
+  // If there isn't one, spawn a new block, add a new block state and put
+  // this block to be scheduled later. Some maintenance of the original block
+  // state is also necessary.
+  if (LBS.isPipelined()) {
+    auto *DedicatedExit = makeDedicatedLoopExit(Loop, BB);
+    if (DedicatedExit == BB) {
+
+      // Trim excedent empty bundles.
+      while (BS.TopInsert.back().empty()) {
```

Comment:
Note: With the latests changes to the post-pipeliner, it seems we can end up with a pipeline of 1 stage, essentially meaning the loop isn't pipelined. It's probably an oversight, and the loop should not have been considered as `isPipelined()`. But still, it makes the `BS.TopInsert.back()` code above crash.

I'd suggest understanding the root cause (you can check `llvm/test/CodeGen/AIE/aie2/schedule/postpipeliner/crash.mir` which now crashed again 😆), and adding an assert in  e.g. `isPipelined()` that if a loop is pipelined, it has non-empty top and bottom inserts.

Reply:
I handled this considering that a swp loop is still valid, but instead of emitting the topinsert, we push a safety margin later on.